packages/next-auth/src/react/index.tsx

```diff
@@ -404,13 +408,16 @@ export function SessionProvider(props: SessionProviderProps) {
     // Set up polling
     if (refetchInterval) {
       const refetchIntervalTimer = setInterval(() => {
-        if (__NEXTAUTH._session) {
+        // TODO: Flip this behavior in next major version
+        const shouldRefetch =
+          props.refetchWhenOffline !== false || navigator.onLine
```

Comment:
We could simplify this one a bit
```suggestion
          props.refetchWhenOffline || navigator.onLine
```